packages/terra-form-select/src/Select.jsx

```diff
@@ -0,0 +1,297 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames/bind';
+import Popup from 'terra-popup';
+import Arrange from 'terra-arrange';
+import IconCaretDown from 'terra-icon/lib/icon/IconCaretDown';
+import 'terra-base/lib/baseStyles';
+import SelectOption from './SelectOption';
+import SelectMenu from './_SelectMenu';
+import SelectUtils from './_SelectUtils';
+import styles from './Select.scss';
+
+const cx = classNames.bind(styles);
+
+const propTypes = {
+  /**
+   * Function to trigger when the user changes the select value.
+   */
+  onChange: PropTypes.func,
+
+  /**
+   * Name of the select field.
+   */
+  name: PropTypes.string,
+
+  /**
+   * Whether the select is required or not.
+   */
+  required: PropTypes.bool,
+
+  /**
+   * The value to start the select on.
+   */
+  defaultValue: PropTypes.string,
+
+  /**
+   * The value of the select element. Use this to create a controlled input.
+   */
+  value: PropTypes.string,
+
+  /**
+   * Select options
+   */
+  children: PropTypes.node.isRequired,
+
+  /**
+   * Indicates if the select is in an invalid state.
+   */
+  isInvalid: PropTypes.bool,
+
+  /**
+   * Indicates if the select should be disabled.
+   */
+  disabled: PropTypes.bool,
+
+  /**
+   * A callback function to let the containing component (e.g. modal) to regain focus.
+   */
+  releaseFocus: PropTypes.func,
+
+  /**
+   * A callback function to request focus from the containing component (e.g. modal).
+   */
+  requestFocus: PropTypes.func,
+};
+
+const defaultProps = {
+  required: false,
+  disabled: false,
+  isInvalid: false,
+};
+
+const contextTypes = {
+  /* eslint-disable consistent-return */
+  intl: (context) => {
+    if (context.intl === undefined) {
+      return new Error('Please add locale prop to Base component to load translations');
+    }
+  },
+};
+
+class Select extends React.Component {
+  constructor(props, context) {
+    super(props, context);
+    this.getInitialState = this.getInitialState.bind(this);
+    this.setTargetRef = this.setTargetRef.bind(this);
+    this.getTargetRef = this.getTargetRef.bind(this);
+    this.handleOnClick = this.handleOnClick.bind(this);
+    this.handleOnKeyDown = this.handleOnKeyDown.bind(this);
+    this.handleOnRequestClose = this.handleOnRequestClose.bind(this);
+    this.handleSelection = this.handleSelection.bind(this);
+    this.wrapOnClick = this.wrapOnClick.bind(this);
+    this.wrapOnKeyDown = this.wrapOnKeyDown.bind(this);
+    this.state = {
+      selectedValue: this.getInitialState(),
+      isOpen: false,
+    };
+  }
+
+  getInitialState() {
+    if (!this.props.value) {
+      const childArray = React.Children.toArray(this.props.children);
+      for (let i = 0; i < childArray.length; i += 1) {
+        const child = childArray[i];
+        if (child.props.isSelected || child.props.value === this.props.defaultValue) {
```

Comment:
It looks like we honor the the `isSelected` prop over the `value` prop over. Would this be the expected behavior?

Reply:
I'm checking the `value` prop at the top of the the function. If the `value` prop is set I'm not setting state at all since it would then be a controlled component. I didn't really have an opinion of honoring one over the other between `isSelected` on the child and `defaultValue` on the parent since they were both serving the same purpose, but I can switch them if anyone has a strong opinion.